src/Core/Services/CipherService.cs

```diff
@@ -1153,6 +1153,8 @@ private async Task EncryptCipherDataAsync(Cipher cipher, CipherView model, Symme
                                 Match = uri.Match
                             };
                             await EncryptObjPropertyAsync(uri, loginUri, new HashSet<string> { "Uri" }, key);
+                            var uriHash = await _cryptoService.HashAsync(uri.Uri, CryptoHashAlgorithm.Sha256);
+                            loginUri.UriChecksum = await _cryptoService.EncryptAsync(uriHash, key);
```

Comment:
🤔 From what I see in `Cipher -> DecryptAsync(...)` the checksum is only taken account when the cipher has a Cipher key. So should a check be added here to see if a cipher key exists in order to set the checksum? Or is it set either way no matter if the cipher key exists or not?

Reply:
Only the validation is tied to the existence of the cipher key, we can set the checksum either way.